GoogleForCreators/web-stories-wp
# Editor: Quick Actions Foundation

## Context

Adds the foundation for Quick Actions.

## Summary

Adding the foundation for Quick Actions.

## Relevant Technical Choices

- Hooked into the highlight api so that we could focus panes and panels across the app. Added benefit of using a familiar api for this future work
- The `TabView` component was holding redundant state for the selected tab. This tab state is already kept in the `LibraryProvider` and `InspectorProvider` so I removed the redundant state from the `TabView` component.
- Added some forward refs where we needed them

## To-do

- [x] add icon menu from #7475 
- [x] add karma tests
- [x] ~position the quick action menu in a good way 😕. LTR doesn't work, but other than that it's ready.~ added in #7510

## User-facing changes

![quick-actions](https://user-images.githubusercontent.com/22185279/118026170-c2e2d080-b31d-11eb-83ac-d4db1db3f3d4.gif)

## Testing Instructions

1. Turn on the `Enable Quick Action Menus` experiment
2. Go to editor
3. Click on the quick action buttons. They should focus specific panels and focus the background element.
    - Note: there is some lag while changing tabs. The animation plays instantly but since there is lag the tab may not flash on the first click. We should fix this in a future ticket.
4. Click the background if not focused, should see the same quick actions. Quick actions should work the same.
5. Add any type of element and focus this element. Quick action menu should hide (will show later once the actions are added for the specific element type).

### QA






1. Turn on the `Enable Quick Action Menus` experiment
2. Go to editor
3. Click on the quick action buttons. They should focus specific panels and focus the background element.
    - Note: there is some lag while changing tabs. The animation plays instantly but since there is lag the tab may not flash on the first click. We should fix this in a future ticket.
4. Click the background if not focused, should see the same quick actions. Quick actions should work the same.
5. Add any type of element and focus this element. Quick action menu should hide (will show later once the actions are added for the specific element type).

### UAT






## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #6151
